src/client.ts

```diff
@@ -42,34 +42,80 @@ export class OpenFeatureClient implements Client {
     this._context = context;
   }
 
-  set logger(logger: Logger) {
+  /**
+   * Sets a logger on the client. This logger supersedes to the global logger
+   * and is passed to various components in the SDK.
+   *
+   * @param {Logger} logger The logger to to be used
+   * @returns {OpenFeatureClient} OpenFeature Client
+   */
+  setLogger(logger: Logger): OpenFeatureClient {
     this._clientLogger = new SafeLogger(logger);
+    return this;
   }
 
-  get logger(): Logger {
-    return this._clientLogger || this.globalLogger();
-  }
-
-  set context(context: EvaluationContext) {
+  /**
+   * Sets client evaluation context that will be used during flag evaluations
+   * on this client.
+   *
+   * @param {EvaluationContext} context Client evaluation context
+   * @returns {OpenFeatureClient} OpenFeature Client
+   */
+  setContext(context: EvaluationContext): OpenFeatureClient {
     this._context = context;
+    return this;
   }
 
-  get context(): EvaluationContext {
+  /**
+   * Access the client evaluation context.
+   *
+   * @returns {EvaluationContext} Client evaluation context
+   */
+  getContext(): EvaluationContext {
     return this._context;
   }
 
-  addHooks(...hooks: Hook<FlagValue>[]): void {
+  /**
+   * Adds client hooks that will run during flag evaluations on this client. Client
```

Comment:
```suggestion
   * Adds hooks that will run during flag evaluations on this client. Client
```

Reply:
same as above. I think it's important to specify that these hooks are scoped to the client. Have to rephrase but I would like to include that information.